### ISSUE TITLE
#DocUpdate: Need inputs for GenPipes troubleshooting (new users)

### ISSUE THREAD

**Issue by shaloo** [c3g/GenPipes_Docs#15]
Please refer to the section: https://genpipes.readthedocs.io/en/latest/get-started/troubleshooting_gp.html

Are there any known issues the new users frequently encounter while using GenPipes:
* Deployed on Compute Canada servers
* Deployed locally on user's workstation 
* Deployed locally in a container / Singularity or docker
* Deployed in GCP cloud

Please note - we plan to include a few of the most frequently encountered issues and the known solutions / workarounds for the same.  This will serve as a self-help / self-unblocking option for new users and encourage them to engage further with GenPipes. Also, it will help reduce email response load for GenPipes support staff helping them to devote their time productively to less mundane and more important GenPipes issues.

What I require:

* Knowledgeable / emails listing issues/solutions
* Forums / other places where I can gather these inputs myself (if possible)

**Comment by rdali** [comment]
The most common issues with CC servers are:
1- not setting up the .bashrc with mugqic modules; jobs fail on the worker nodes
2- not setting up the CCRI (RAP_ID) correctly

For issues with Docker and GCP cloud, Pierre Olivier is the best person to consult. 
@pphector can you ask PO and Ed to list any issues users have reported repeatedly?

@shaloo Most of the issues come to private email support that we run. I hope Hector/Ed/PO can give you an idea of the most frequent request. There is a public GenPipes google group which you can also look into: https://groups.google.com/forum/?utm_medium=email&utm_source=footer#!forum/genpipes

**Comment by shaloo** [comment]
Thanks Rola for your prompt response!  Looking forward to hearing from Hector and Pierre about the Docker and GCP cloud ones.

Saw the Google forum, found 1-2 good ones there (Qsub command not found error etc.), will include those for now.

Just remembered seeing some teething issues on "GSoD/GSoC GenPipes" slack channel during May 19. I guess those were mostly related to CC account access so I will utilise that as well.
I am assuming there is no other similar “GenPipes" slack channel (Not GSoC one) where I could gather more material for docs - Getting Started troubleshooting guide section?

best,
Shaloo

On 19-Jul-2019, at 7:20 PM, Rola Dali <notifications@github.com<mailto:notifications@github.com>> wrote:


The most common issues with CC servers are:
1- not setting up the .bashrc with mugqic modules; jobs fail on the worker nodes
2- not setting up the CCRI (RAP_ID) correctly

For issues with Docker and GCP cloud, Pierre Olivier is the best person to consult.
@pphector<https://github.com/pphector> can you ask PO and Ed to list any issues users have reported repeatedly?

@shaloo<https://github.com/shaloo> Most of the issues come to private email support that we run. I hope Hector/Ed/PO can give you an idea of the most frequent request. There is a public GenPipes google group which you can also look into: https://groups.google.com/forum/?utm_medium=email&utm_source=footer#!forum/genpipes

**Comment by pphector** [comment]
Hi Shaloo,

Sorry for the late reply, both Pierre and myself are on vacation and in my
case I have been camping with limited internet access. I am just getting up
to date on this.

In terms of GenPipes troubleshooting issues with Docker and the cloud, I
haven't used it that much, so I don't really know any issues off the top of
my head. But I think that the main thing will be to emphasize that GenPipes
needs to be run in batch mode, so maybe we can put something like that.
Also, I remember Pierre Olivier mentioning that sometimes STAR (from the
rnaseq pipelines) gave him error messages, but I'm not entirely sure what
they were exactly.

Another common troubleshooting issue that many of our users encounter has
to do with missing genomes/annotations. In those cases, they have to get
the missing information themselves or generate the specific file that is
needed. Maybe it would be nice to have a list of genome files and
annotations and where they come from so people can look and see if they are
missing anything.

Apologies once again for the delay, I officially return to the office on
Monday and I will take another look at this and see if anyone else has
anything to add.

Best,
Jose Hector Galvez

**Comment by shaloo** [comment]
Thanks Hector.

On 24-Jul-2019, at 9:38 PM, J. Hector Galvez <notifications@github.com<mailto:notifications@github.com>> wrote:

Hi Shaloo,

Sorry for the late reply, both Pierre and myself are on vacation and in my
case I have been camping with limited internet access. I am just getting up
to date on this.

In terms of GenPipes troubleshooting issues with Docker and the cloud, I
haven't used it that much, so I don't really know any issues off the top of
my head. But I think that the main thing will be to emphasize that GenPipes
needs to be run in batch mode, so maybe we can put something like that.
Also, I remember Pierre Olivier mentioning that sometimes STAR (from the
rnaseq pipelines) gave him error messages, but I'm not entirely sure what
they were exactly.

Another common troubleshooting issue that many of our users encounter has
to do with missing genomes/annotations. In those cases, they have to get
the missing information themselves or generate the specific file that is
needed. Maybe it would be nice to have a list of genome files and
annotations and where they come from so people can look and see if they are
missing anything.

Apologies once again for the delay, I officially return to the office on
Monday and I will take another look at this and see if anyone else has
anything to add.

Best,
Jose Hector Galvez

**Comment by shaloo** [comment]
References:

* Logs - first time GenPipes deployment / CC access data that I saved in May19  (Doc Added)

*out of memory issues https://groups.google.com/forum/#!topic/GenPipes/7QskD_lL5I0
RNA seq pipeline issue (faced by more than 1 user) that required STAR parameter fix and buffer sizing see https://groups.google.com/forum/#!topic/GenPipes/EC2VeLz3i0Y (documentation updated)
